contracts/AKAP.sol

```diff
@@ -56,24 +56,36 @@ contract AKAP is ERC721Full {
     }
 
     function claim(uint parentId, bytes calldata label) external returns (uint status) {
-        // Make sure parent is the special root parent or caller is owner of/approved on parent
-        require(parentId == 0x0 || _isApprovedOrOwner(_msgSender(), parentId), "AKAP: invalid parent ownership");
+        // Claim logic is as follows:
+
+        // Case 1:
+        // A node that does exist can be extended if the _msgSender() is the owner of nodeId.
+
+        // Case 2:
+        // A node that does not exist can be claimed if the _msgSender() is the owner of parentId.
+        // If parentId is the special case 0x0, you can consider the _msgSender() as the "owner" of parentId.
+
+        // Case 3:
+        // A node that does exists but is expired will be transferred to the new _msgSender(). This still
+        // assumed that the _msgSender() is the parentId owner, including special case 0x0 as above.
 
         // Get hash/id of the node caller is claiming
         uint nodeId = hashOf(parentId, label);
 
+        bool isParentOwner = parentId == 0x0 || _isApprovedOrOwner(_msgSender(), parentId);
+
         if (_exists(nodeId) && _isApprovedOrOwner(_msgSender(), nodeId)) {
             // Caller is current owner/approved, extend lease..
             nodes[nodeId].expiry = now + 52 weeks;
 
             return 1;
-        } else if (!_exists(nodeId)) {
+        } else if (!_exists(nodeId) && isParentOwner) {
             // Node does not exist, allocate to caller..
```

Comment:
We maybe need to reconsider moving this to a a separate function. The special case of parentId = 0x0, can also be seen as base case nodeId is assigned and has no owner.

Reply:
Not clear on what should be considered moving? The isParentOwner check?

Reply:
The whole case 2 block, and restrict claim to onlyApproved and existing. Still need to account for node expired case though. 

function claimNew(parentId, label) external NonExisting {
   // Case 2 here
}

Reply:
Not keen on having multiple claim functions as that would force the caller to apply logic prior to making a call, and that becomes tricky in an asynchronous/eventually consistent world.

Better to encapsulate such logic within the claim function, that way the caller knows they can simply use that one function without thinking about existing state.